tests/containers/bci_tests.pm

```diff
@@ -0,0 +1,135 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: bci-tests runner
+#   SUSE Linux Enterprise Base Container Images (SLE BCI)
+#   provides truly open, flexible and secure container images and application
+#   development tools for immediate use by developers and integrators without
+#   the lock-in imposed by alternative offerings.
+#   - install needed python environment
+#   - clone bci-tests repo
+#   - build project
+#   - run test for each given environment
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use Mojo::Base qw(consoletest);
+use XML::LibXML;
+use utils qw(systemctl zypper_call ensure_ca_certificates_suse_installed);
+use version_utils qw(get_os_release);
+use containers::common;
+use testapi;
+
+sub run {
+    my ($self) = @_;
+    $self->select_serial_terminal;
+
+    my $runtime        = get_required_var('RUNTIME');
+    my $bci_tests_repo = get_required_var('BCI_TESTS_REPO');
+    my $test_envs      = get_required_var('BCI_TEST_ENVS');
+    my $bci_devel_repo = get_var('BCI_DEVEL_REPO');
+    my $bci_timeout    = get_var('BCI_TIMEOUT', 900);
+
+    ensure_ca_certificates_suse_installed;
+
+    my ($running_version, $sp, $host_distri) = get_os_release;
+    if ($runtime eq 'podman') {
+        install_podman_when_needed($host_distri);
+        install_buildah_when_needed($host_distri);
+    }
+    elsif ($runtime eq 'docker') {
+        install_docker_when_needed($host_distri);
+    }
+    else {
+        die("Runtime $runtime not given or not supported");
+    }
+
+    record_info('Install', 'Install needed packages');
+    zypper_call('--quiet in git-core python39', timeout => 600);
```

Comment:
hardcoding certain python version ? there should be some good reasons for this . do we have them ? can we add some comment here regarding this ?

Reply:
It's a PoC, and this is to be investigated in [this ticket](https://progress.opensuse.org/issues/98183) 
It comes from [requierements of the project](https://gitlab.suse.de/dancermak/bci-tests#what-do-i-need-to-contribute). But I agree with you, but this is a working PoC, and this is to be investigated in [this ticket](https://progress.opensuse.org/issues/98183) when we need to enable it on other hosts.

Reply:
fare enough